src/epost/Upload.tsx

```diff
@@ -0,0 +1,434 @@
+  senderCity: string;
+}
+
+interface InitialFormData extends FormData {
+  pdfFile: File | null;
+}
+
+export interface EPostFormData extends FormData {
+  pdfFile: File;
+}
+
+const useStyles = makeStyles((theme) => ({
+  wrapper: {
+    paddingTop: theme.spacing(8),
+    width: 'auto',
+    display: 'block', // Fix IE 11 issue.
+    [theme.breakpoints.up(400 + theme.spacing(6))]: {
+      width: 800,
+      marginLeft: 'auto',
+      marginRight: 'auto',
+    },
+  },
+  form: {
+    width: '100%', // Fix IE 11 issue.
+    marginTop: theme.spacing(),
+  },
+  formHidden: {
+    display: 'none',
+  },
+  inputBase: {
+    marginLeft: '10px',
+    ml: 10,
+    flex: 1,
+  },
+  submit: {
+    margin: '10px 0 50px',
+  },
+  fileInput: {
+    display: 'none',
+  },
+  subtitle: {
+    paddingTop: '20px',
+  },
+  subHeader: {
+    marginTop: '30px',
+  },
+  uploadBox: {
+    border: '2px solid #479f9d',
+    padding: '20px',
+    margin: '20px 0 20px 0',
+  },
+  fileName: { fontWeight: 'bold' },
+  italicText: {
+    fontStyle: 'italic',
+  },
+  rowItem: { display: 'flex', flex: 1 },
+  expandRowSymbol: {
+    display: 'flex',
+    flex: 1,
+    alignItems: 'flex-end',
+    justifyContent: 'flex-end',
+  },
+  collapsibleRow: {
+    cursor: 'pointer',
+    backgroundColor: '#f1f1f1',
+    padding: 10,
+    display: 'flex',
+    flexDirection: 'row',
+    marginTop: 30,
+  },
+  row: { display: 'flex', justifyContent: 'space-between' },
+  modal: { padding: '20px' },
+  modalButton: { marginTop: '20px', width: '100%', display: 'block' },
+}));
+
+const Upload: React.FC<RouteComponentProps<{
+  token?: string;
+}>> = () => {
+  const fileInput = React.useRef<HTMLInputElement>(null);
+
+  const blankForm = {
+    pdfFile: null,
+    addressLineOne: '',
+    addressLineTwo: '',
+    postCode: '',
+    city: '',
+    senderAddressLineOne: 'HiDoc Technologies GmbH',
+    senderAddressLineTwo: 'Hohe Bleichen 22',
+    senderPostCode: '20354',
+    senderCity: 'Hamburg',
+  };
+
+  const styles = useStyles();
+  const [formData, setFormData] = useState<InitialFormData>(blankForm);
+  const [formHidden, setFormHidden] = useState<boolean>(true);
+  const [showSuccessMessage, setShowSuccessMessage] = useState(false);
+  const [success, setSuccess] = useState<Success | null>(null);
+  const [error, setError] = useState('');
+
+  const toggleSenderDetails = useCallback(() => {
+    setFormHidden(!formHidden);
+  }, [formHidden]);
+
+  const handleFileInputChange = useCallback(() => {
+    if (
+      fileInput.current &&
+      fileInput.current.files !== null &&
+      fileInput.current.files.length
+    ) {
+      setError('');
+      return setFormData({ ...formData, pdfFile: fileInput.current.files[0] });
+    }
+    return;
+  }, [formData, fileInput]);
+
+  const handleSubmit = useCallback(
+    (event: { preventDefault: () => void }) => {
+      event.preventDefault();
+      const ePostForm = document.getElementById('ePostForm');
```

Comment:
We need to reset the error before the call, so that the user can see that there is a new error.
```suggestion
      setError('');
      const ePostForm = document.getElementById('ePostForm');
```

Reply:
Applied ✅